src/Symfony/Bridge/Doctrine/Form/ChoiceList/ORMQueryBuilderLoader.php

```diff
@@ -62,7 +62,7 @@ public function getEntitiesByIds($identifier, array $values)
             $metadata = $this->queryBuilder->getEntityManager()->getClassMetadata(current($this->queryBuilder->getRootEntities()));
 
             foreach ($this->getEntities() as $entity) {
-                if (\in_array(current($metadata->getIdentifierValues($entity)), $values, true)) {
+                if (\in_array((string) current($metadata->getIdentifierValues($entity)), $values, true)) {
```

Comment:
what about setting the last argument to false instead?

Reply:
I've wrote it this way with the following logic in mind: if we can optimize the query, it's because we were able to cast ids to strings in the first place, so submitted values should be strictly equivalent to ids casted to strings again. That said, I don't really care :), so let me know if you'd rather have me change it.

Reply:
You mean that `$values` can only contain strings, for the case we target?

Reply:
Yes, the loader is called from the choices to values transformers, and here we handle submission of strings.